### PR TITLE
Lorenz transformation - physics

### PR DESCRIPTION
**Pull request by avivfaraj**
### Describe your change:
Implemented Lorenz transformation using four vector. 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: physics/lorenz_transformation_four_vector.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import numpy as np  # type: ignore
+from sympy import symbols  # type: ignore
+
+# Coefficient
+# Speed of light (m/s)
+c = 299792458
+
+
+# Vehicle's speed divided by speed of light (no units)
+def beta(u: float) -> float:
```
[the remark this answers is not in the document]
Comment (avivfaraj) [reply physics/lorenz_transformation_four_vector.py]: changed u to velocity

**Comment by avivfaraj** [comment]
pre-commit fails because of spelling of a word in the URL of the reference. The URL contains Relativ instead of Relative, so correcting spelling will break the link

File: physics/lorenz_transformation_four_vector.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+    >>> gamma(2*c)
+    Traceback (most recent call last):
+      ...
+    ValueError: Speed must not exceed Light Speed 299,792,458 [m/s]!
+    """
+    return 1 / (sqrt(1 - beta(velocity) ** 2))
+
+
+def transformation_matrix(velocity: float) -> np.array:
```
[the remark this answers is not in the document]
Comment (avivfaraj) [reply physics/lorenz_transformation_four_vector.py:103]: I am not sure why I keep getting this message. There are tests listed in this function

File: physics/lorenz_transformation_four_vector.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    """
+    >> transformation_matrix(29979245)
```
Comment (Rohanrbharadwaj) [inline physics/lorenz_transformation_four_vector.py:105]:
```suggestion
    >>> transformation_matrix(29979245)
```
3, not 2

Comment (avivfaraj) [reply physics/lorenz_transformation_four_vector.py]: Thank you so much! 
Can't believe I missed that

Comment (Rohanrbharadwaj) [reply physics/lorenz_transformation_four_vector.py]: Some tests in the `beta` and `gamma` functions would require the same change, although the bot seems pleased.

**Review by Rohanrbharadwaj — APPROVED** [review]

**Review by poyea — APPROVED** [review]
Thank you for your pull request and initializing `physics`!🤩

**Pull request merged**